lib/ember-qunit/qunit-module.js

```diff
@@ -45,21 +35,24 @@ export function createModule(Constructor, name, description, callbacks) {
   var module = new Constructor(name, description, callbacks);
 
   qunitModule(module.name, {
-    setup: function(assert) {
-      var done = assert.async();
-      return module.setup().then(function() {
+    beforeEach() {
+      return module.setup(...arguments).then(() => {
+        Object.keys(module.context).forEach((key) => {
+          this[key] = module.context[key];
+        });
+
         if (beforeEach) {
-          beforeEach.call(module.context, assert);
+          beforeEach.apply(this, arguments);
```

Comment:
Lets swap this to:

```js
beforeEach.apply(module.context, arguments);
```

Once https://github.com/rwjblue/ember-qunit/pull/237 is landed the `module.context` is the right thing anyways...